extensions/nns/src/install_nns.rs

```diff
@@ -520,37 +488,24 @@ pub fn upload_nns_sns_wasms_canister_wasms(dfx_cache_path: &Path) -> anyhow::Res
         ..
     } in SNS_CANISTERS
     {
-        let sns_cli = bundled_binary(dfx_cache_path, "sns")?;
-        let wasm_path = nns_wasm_dir(dfx_cache_path).join(wasm_name);
-        let mut command = Command::new(sns_cli);
-        command
-            .arg("add-sns-wasm-for-tests")
-            .arg("--network")
-            .arg("local")
-            .arg("--override-sns-wasm-canister-id-for-tests")
-            .arg(NNS_SNS_WASM.canister_id)
-            .arg("--wasm-file")
-            .arg(&wasm_path)
-            .arg(upload_name);
-        command
-        .stdin(process::Stdio::null())
-        .output()
-            .map_err(anyhow::Error::from)
-            .and_then(|output| {
-                if output.status.success() {
-                    Ok(())
-                } else {
-                    Err(anyhow!(
-                        "Failed to upload {} from {} to the nns-sns-wasm canister:\n{:?} {:?}\nStdout:\n{:?}\n\nStderr:\n{:?}",
-                        upload_name,
-                        wasm_path.to_string_lossy(),
-                        command.get_program(),
-                        command.get_args(),
-                        String::from_utf8_lossy(&output.stdout),
-                        String::from_utf8_lossy(&output.stderr)
-                    ))
-                }
-            })?;
+        let wasm_path = nns_wasm_dir(dfx_cache_path)
+            .join(wasm_name)
+            .display()
+            .to_string();
+        let args = vec![
+            "add-sns-wasm-for-tests",
+            "--network",
+            "local",
+            "--override-sns-wasm-canister-id-for-tests",
+            NNS_SNS_WASM.canister_id,
+            "--wasm-file",
+            &wasm_path,
+            upload_name,
+        ];
+        call_extension_bundled_binary("sns-cli", &args)
+            .map_err(|e| anyhow!(
+                        "Failed to upload {upload_name} from {wasm_path} to the nns-sns-wasm canister by calling `sns-cli` with args {args:?}: {e}"
+                    ))?;
```

Comment:
I think this is how we could get rid of the `sns-cli` dependency in dfx nns extension. It still requires us to pull in a library to get the sha256sum digest of the wasms, but that should be trivial.

```
/// Uploads wasms to the nns-sns-wasm canister.
#[context("Failed to upload wasm files to the nns-sns-wasm canister; it may not be possible to create an SNS.")]
pub fn upload_nns_sns_wasms_canister_wasms(dfx_cache_path: &Path, nns_url: &Url) -> anyhow::Result<()> {
    for SnsCanisterInstallation {
        upload_name,
        wasm_name,
        ..
    } in SNS_CANISTERS
    {
        let wasm_path = nns_wasm_dir(dfx_cache_path)
            .join(wasm_name)
            .display()
            .to_string();
        let summary = format!(
            "Proposal to Publish the SNS {} Canister WASM to SNS-W",
            upload_name,
        );
        let args = vec![
            "--nns-url",
            nns_url.as_str(),
            "propose-to-add-wasm-to-sns-wasm",
            "--test-neuron-proposer",
            "--canister-type",
            upload_name,
            "--wasm-module-path",
            &wasm_path,
            "--wasm-module-sha256",
            "", // TODO - get the sha256sum of the wasm file thats been downloaded by dfx_cache
            "--summary",
            &summary,
        ];
        call_extension_bundled_binary("ic-admin", &args)
            .map_err(|e| anyhow!(
                        "Failed to upload {upload_name} from {wasm_path} to the nns-sns-wasm canister by calling `ic-admin` with args {args:?}: {e}"
                    ))?;
    }
    Ok(())
}
```

Reply:
It probably not worth doing in this PR, but I wanted to spend the time to write this out for posterity :D

Reply:
oh, that's pretty cool you've looked into this! Having `sns-cli` in both `nns` and `sns` ext is the only PITA here ^^

As you've said, let's try to remove it in the next PR, just note you'd need to be able to figure out if it still works correctly (or if such change it's already covered by e2e tests) cause I'd have no clue 🥲